main/inc/lib/hook/HookManagement.class.php

```diff
@@ -0,0 +1,516 @@
+<?php
+/* For licensing terms, see /license.txt */
+/**
+ * @TODO: Improve description
+ * @package chamilo.hookmanagement
+ */
+
+class HookManagement implements HookManagementInterface
+{
+    /**
+     * Constructor
+     */
+    protected function __construct()
+    {
+        $this->tables[TABLE_PLUGIN_HOOK_OBSERVER] = Database::get_main_table(TABLE_PLUGIN_HOOK_OBSERVER);
+        $this->tables[TABLE_PLUGIN_HOOK_EVENT] = Database::get_main_table(TABLE_PLUGIN_HOOK_EVENT);
+        $this->tables[TABLE_PLUGIN_HOOK_CALL] = Database::get_main_table(TABLE_PLUGIN_HOOK_CALL);
+
+        $this->hookCalls = $this->listAllHookCalls();
+        $this->hookEvents = $this->listAllHookEvents();
+        $this->hookObservers = $this->listAllHookObservers();
+    }
+
+    /**
+     * Instance the hook manager
+     * @staticvar null $result
+     * @return HookManagement
+     */
+    static function create()
+    {
+        static $result = null;
+
+        return $result ? $result : $result = new self();
+    }
+
+    /**
+     * Install the plugin
+     * @return void
+     */
+    public function install()
+    {
+        $this->installDatabase();
+        $this->initDatabase();
+    }
+
+    /**
+     * Uninstall the plugin
+     * @return void
+     */
+    public function uninstall()
+    {
+        $this->uninstallDatabase();
+    }
+
+    /**
+     * Create the database tables for the plugin
+     * @return void
+     */
+    private function installDatabase()
+    {
+        $sql = 'CREATE TABLE IF NOT EXISTS ' . $this->tables[TABLE_PLUGIN_HOOK_OBSERVER] . '( ' .
```

Comment:
Actually, I see that these *already are* in db_main.sql further down this page, so this really has to be deleted.